shelm

```diff
@@ -68,10 +68,12 @@ json_location() {
 fetch_github_release() {
 	pkgname="$1"
 	version="$2"
-	curlargs=-sSL # no progress, show errors, follow redirects
+	curlargs=-sSLf # no progress, show errors, follow redirects, exit with error on HTTP status >=400
 	url="https://github.com/$pkgname/archive/$version.tar.gz"
 	echo "fetching $url"
-	curl "$curlargs" "$url" | tar -xz
```

Comment:
How about leaving this line as is and adding `set -o pipefail`? It would be nice to avoid dealing with a temporary file.

Reply:
(There's a fix to the `ls */*/*` globbing needed to get pipefail to work, on it.)